partner_operating_unit/tests/test_res_partner_operating_unit.py

```diff
@@ -0,0 +1,99 @@
+##############################################################################
+#
+#    Copyright (c) 2019 brain-tec AG (http://www.braintec-group.com)
+#    License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+#
+##############################################################################
+
+from odoo.tests import common
+
+
+class TestResPartnerOperatingUnit(common.TransactionCase):
+
+    def setUp(self):
+        super(TestResPartnerOperatingUnit, self).setUp()
+
+        self.ResUsers = self.env['res.users']
+        self.ResPartner = self.env['res.partner']
+        self.OperatingUnit = self.env['operating.unit']
+
+        # Companies.
+        self.company = self.env.ref('base.main_company')
+
+        # Groups.
+        self.group_partner_manager = self.env.ref('base.group_partner_manager')
+
+        # Operating Units.
+        self.opunit1 = self.OperatingUnit.create({
+            'name': 'Operating Unit #1',
+            'code': 'OP#1',
+            'partner_id': self.env.ref('base.main_partner').id,
+        })
+        self.opunit2 = self.OperatingUnit.create({
+            'name': 'Operating Unit #2',
+            'code': 'OP#2',
+            'partner_id': self.env.ref('base.main_partner').id,
+        })
+
+        # Partners.
+        self.partner1 = self.env.ref('base.res_partner_1')
+        self.partner2 = self.env.ref('base.res_partner_2')
+        self.partner3 = self.env.ref('base.res_partner_3')
+
+        # Sets the operating-units for each partner.
+        self.partner1.operating_unit_ids = [(6, False, [self.opunit1.id])]
+        self.partner2.operating_unit_ids = [(6, False, [self.opunit2.id])]
+        self.partner3.operating_unit_ids = [(6, False, [self.opunit1.id,
+                                                        self.opunit2.id])]
+
+        # Users.
+        self.user1_id = self._create_user('user_1', self.group_partner_manager,
+                                          [self.opunit1])
+        self.user2_id = self._create_user('user_2', self.group_partner_manager,
+                                          [self.opunit2])
+
+    def _create_user(self, login, groups, operating_units):
+        """ Create a user.
+        """
+        user = self.ResUsers.with_context({'no_reset_password': True}).create({
+            'name': 'Demo User',
+            'login': login,
+            'password': 'demo',
+            'email': '{0}@yourcompany.com'.format(login),
+            # 'company_id': company.id,
```

Comment:
Why is this commented? If unneeded, please remove it

Reply:
Thanks. I'll remove it, and tell you when done.

Reply:
The lines were removed. I also removed the Python constraint for ```res.partner.category``` since that model doesn't have link to a company.

Reply:
Ok make sense. Great